### DISCUSSION TITLE
Appling ERGA to environmental data

### DISCUSSION THREAD

**Discussion by Antonio-Longo** [ERGA-consortium/pipelines#43] [Q&A]
Hi everybody,

I am Antonio, from  Naples (Italy) and i wish you all an happy (hopefully) new year. 
As i mentioned above, i am trying to apply your pipeline to environmental data, 
now is running but I am having some issues with the transposonPSI rule (where 
I had to change some commands since gunzip was stopping the workflow to ask
preferences). There are issues also with split uniprot rule.

[completelog.txt](https://github.com/ERGA-consortium/pipelines/files/13854573/completelog.txt)

i attached the complete log of snakemake to let you have an idea of what is going
here. 

Do you have any idea?
Thanks a lot,
Antonio

**Comment by tbrown91** [comment]
Hi Antonio,

I would first ask if you have the space on your disk to perform all of these tasks? I see that you are running on a desktop and you are writing any temp files to /tmp - do you have an idea how much space you have available?

It looks from your log file like the split rule is generally working and then one of the transposonPSI jobs is failing and cleaning up all of the intermediate files.

Are you able to run a single chunk manually just to make sure they are finishing and the conda environments are set up correctly?

@sdind if you have any ideas as well, please jump in

**Comment by Antonio-Longo** [comment]
Hi, 

at the moment your pipeline and my data are on a 4 tB hdd, while the tmp has about 160 gb of free space. I suppose that the warning message (that i attached) is related to -t (threads?) of blastpgp command from transposonPSI rule. I tried to times and in both cases there were outputs  for the rule transposonPSI, but suddenly both crashed.

[warning.txt](https://github.com/ERGA-consortium/pipelines/files/13903548/warning.txt)

as you can see there is a strange pattern in the warning messages from the log here:

![image](https://github.com/ERGA-consortium/pipelines/assets/108012591/74af4635-94d7-4318-9e7b-ebd0bfd25a98)


thanks again!

**Comment by tbrown91** [comment]
Hi Antonio,
I wouldn't worry about these warnings, the jobs should still run. The -t argument is also a default of `blastpgp` inside transposonPSI, which is itself run in default mode. In case you are interested, this argument is: `-t  Use composition based score adjustment`.

Do you see any of the expected output files? These should be called something like:
`uniprot_sprot_chunk1.fa.TPSI.topHits`
The cleanup of the `split.fa` and `*.allHits` files is expected from a correct run, as these files are not required for the next job.

I just tested a chunk on our system and it ran ok with the commands:
```
gaas_fasta_splitter.pl -f files/uniprot_sprot.fasta --nb_chunks 10000 -o ./files/split_result2
transposonPSI.pl files/split_result2/uniprot_sprot_chunk1.fa prot
```
and I got all of these warnings and it did a lot of writing and deleting of temp files in my directory.

My next guess would be that you are running the snakemake pipeline with too many cores (`snakemake --cores=n`), or that the read/write is too fast for your hard drive. If it is the latter case, you should be able to keep running the pipeline and it will finish all of these jobs. 

I hope there's some helpful information in here somewhere.

Tom
